src/Persistence/PersistentObjectFactory.php

```diff
@@ -0,0 +1,471 @@
+    {
+        return static::repository()->findBy($attributes);
+    }
+
+    /**
+     * @param Parameters $attributes
+     *
+     * @return T&Proxy
+     * @phpstan-return Proxy<T>
+     */
+    final public static function random(array $attributes = []): object
+    {
+        return static::repository()->random($attributes);
+    }
+
+    /**
+     * @param Parameters $attributes
+     *
+     * @return T&Proxy
+     * @phpstan-return Proxy<T>
+     */
+    final public static function randomOrCreate(array $attributes = []): object
+    {
+        try {
+            return static::repository()->random($attributes);
+        } catch (\RuntimeException) {
+            return static::new()->create($attributes);
+        }
+    }
+
+    /**
+     * @param Parameters $attributes
+     *
+     * @return list<T&Proxy>
+     * @phpstan-return list<Proxy<T>>
+     */
+    final public static function randomSet(int $number, array $attributes = []): array
+    {
+        return static::repository()->randomSet($number, $attributes);
+    }
+
+    /**
+     * @param Parameters $attributes
+     *
+     * @return list<T&Proxy>
+     * @phpstan-return list<Proxy<T>>
+     */
+    final public static function randomRange(int $min, int $max, array $attributes = []): array
+    {
+        return static::repository()->randomRange($min, $max, $attributes);
+    }
+
+    /**
+     * @param Parameters $attributes
+     */
+    final public static function count(array $attributes = []): int
+    {
+        return static::repository()->count($attributes);
+    }
+
+    final public static function truncate(): void
+    {
+        static::repository()->truncate();
+    }
+
+    /**
+     * @phpstan-return RepositoryProxy<T>
+     */
+    final public static function repository(): RepositoryProxy
+    {
+        return static::persistenceManager()->repositoryFor(static::class());
+    }
+
+    /**
+     * @return T&Proxy
+     * @phpstan-return Proxy<T>
+     */
+    final public function create(array|callable $attributes = []): object
+    {
+        [$object, $parameters, $postPersistCallbacks] = $this->normalizeAndInstantiate($attributes);
+
+        $object = new Proxy($object);
+
+        if (!$this->isPersisting()) {
+            return $object;
+        }
+
+        if ($this->cascadePersist && !$postPersistCallbacks) {
+            return $object;
+        }
+
+        return $object
+            ->save()
+            ->withoutAutoRefresh(function (Proxy $proxy) use ($parameters, $postPersistCallbacks): void {
+                $callbacks = [...$postPersistCallbacks, ...$this->afterPersist];
+
+                if (!$callbacks) {
+                    return;
+                }
+
+                foreach ($callbacks as $callback) {
+                    $proxy->executeCallback($callback, $parameters);
+                }
+
+                $proxy->save(); // save again as afterPersist events may have modified
+            });
```

Comment:
I kept the "old" way to do this... I saw you've changed that in foundry-next, we still can do this in another PR